router/router.go

```diff
@@ -1505,10 +1521,30 @@ func (rt *HandleT) commitStatusList(responseList *[]jobResponseT) {
 		workspaceID := resp.status.WorkspaceId
 		eventName := gjson.GetBytes(resp.JobT.Parameters, "event_name").String()
 		eventType := gjson.GetBytes(resp.JobT.Parameters, "event_type").String()
-		key := fmt.Sprintf("%s:%s:%s:%s:%s:%s:%s", parameters.SourceID, parameters.DestinationID, parameters.SourceBatchID, resp.status.JobState, resp.status.ErrorCode, eventName, eventType)
+		key := fmt.Sprintf(
+			"%s:%s:%s:%s:%s:%s:%s",
+			parameters.SourceID,
+			parameters.DestinationID,
+			parameters.SourceBatchID,
+			resp.status.JobState,
+			resp.status.ErrorCode,
+			eventName,
+			eventType,
+		)
 		_, ok := connectionDetailsMap[key]
 		if !ok {
-			cd := utilTypes.CreateConnectionDetail(parameters.SourceID, parameters.DestinationID, parameters.SourceBatchID, parameters.SourceTaskID, parameters.SourceTaskRunID, parameters.SourceJobID, parameters.SourceJobRunID, parameters.SourceDefinitionID, parameters.DestinationDefinitionID, parameters.SourceCategory)
+			cd := utilTypes.CreateConnectionDetail(
+				parameters.SourceID,
+				parameters.DestinationID,
+				parameters.SourceBatchID,
+				parameters.SourceTaskID,
+				parameters.SourceTaskRunID,
+				parameters.SourceJobID,
+				parameters.SourceJobRunID,
+				parameters.SourceDefinitionID,
+				parameters.DestinationDefinitionID,
+				parameters.SourceCategory,
+			)
```

Comment:
[minor] I think it is safer and more readable to create the struct, rather than use a function given the number of arguments we need to pass.

Reply:
Sure. The same with statusDetail as well.
Let's take this up in another PR because it'll introduce changes in too many places which might cause oversight.